pkg/server/server.go

```diff
@@ -327,25 +327,23 @@ func (s *Server) Run(ctx context.Context) error {
 		return err
 	}
 
-	if s.Options.Cache.Enabled {
-		if err := s.AddPostStartHook("kcp-start-optional-informers", func(hookContext genericapiserver.PostStartHookContext) error {
-			// TODO(p0lyn0mial): failing the optional hook should not render the main server unhealthy
-			logger := logger.WithValues("postStartHook", "kcp-start-optional-informers")
-			s.CacheKcpSharedInformerFactory.Start(hookContext.StopCh)
-			s.CacheKcpSharedInformerFactory.WaitForCacheSync(hookContext.StopCh)
+	if err := s.AddPostStartHook("kcp-start-optional-informers", func(hookContext genericapiserver.PostStartHookContext) error {
+		// TODO(p0lyn0mial): failing the optional hook should not render the main server unhealthy
```

Comment:
We can remove this comment, can't we?

Reply:
Or does this depend on `s.Options.Cache.KubeconfigFile` being set?

Reply:
>  the main server should not wait for the cache server - there is plenty of work it could do without the cache

the logic has not changed with this PR. As per other comments the controller for APIExportEndpointSices reconciliation would need the informers related to the cache server to have synced. It may not be the case of other controllers.